src/crypto/public_key/rsa.js

```diff
@@ -31,6 +31,9 @@ import random from '../random';
 import config from '../../config';
 import util from '../../util';
 
+const nodeCrypto = util.getNodeCrypto(); // for native node keygen
```

Comment:
The reason `getWebCryptoAll` is called inside the function, is so that you can change its behavior using `config.use_native`. Could you do the same thing with `getNodeCrypto`?

Reply:
Sure. That means I'd run `asn1.define('RSAPrivateKey'` on top regardless of `config.use_native` setting, then do `getNodeCrypto` inside `generate` so that it's sensitive to `config.use_native` which user may have set sometime after the whole lib was initiated. Did I understand correctly?

This is not a nitpick, I think it's very much expected that keygen stays non-native when `config.use_native` is false.

Reply:
Sure. That means I'd run `asn1.define('RSAPrivateKey'` on top regardless of `config.use_native` setting, then do `getNodeCrypto` inside `generate` so that it's sensitive to `config.use_native` which user may have set sometime after the whole lib was initiated. Did I understand correctly?

This is not a nitpick, I think it's very much expected that keygen stays non-native when `config.use_native` is false.

Reply:
Oh, right. You could do both actually (call the function in both places, here to check whether we're on node and inside the function to check `config.use_native`).

Reply:
Yup, that's what I was thinking. Will do.